generator/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+type Event struct {
+	Status int
+}
+
+func setInterval(function func(), seconds int) {
+	ticker := time.NewTicker(time.Duration(seconds) * time.Second)
+	done := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				function()
+			case <-done:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func main() {
+	setInterval(GenerateWindEvents, 5)
+	http.ListenAndServe(":1234", nil)
+}
+
+func GenerateWindEvents() {
+	if checkForStatus() != 1 {
+		return
+	}
+
+	const (
+		minPower float64 = 2
+		maxPower float64 = 120
+	)
+
+	var mean float64 = randMean(getLastSpeed())
+	if mean == -1 {
+		return
+	}
+	var windPower float64 = math.Round(normPower(mean)*10) / 10
+	var beaufortValue uint8 = uint8(math.Round(windPower / 10))
+	if beaufortValue == 0 {
+		beaufortValue = 1
+	}
+
+	reqBody, err := json.Marshal(map[string]interface{}{
+		"power":     windPower,
+		"direction": randDirection(),
+		"beaufort":  beaufortValue,
+		"Event": map[string]string{
+			"device_type": "wind",
+		},
+	})
+
+	if err != nil {
+		fmt.Println("Could not convert to json")
+	}
+
+	req, err := http.Post("http://localhost:8080/wind/event", "application/json", bytes.NewBuffer(reqBody))
+
+	if err != nil {
+		fmt.Println("Cannot create wind event", err)
+	} else {
+		fmt.Println("Wind event created")
+		defer req.Body.Close()
```

Comment:
If it's the last line - defer is unneeded; `defer` means that we will call this function when the main scope function (in which it was called) will finish